cheroot/ssl/pyopenssl.py

```diff
@@ -320,20 +333,18 @@ def __init__(
 
         self._environ = None
 
-    def bind(self, sock):
-        """Wrap and return the given socket."""
-        if self.context is None:
-            self.context = self.get_context()
-        conn = SSLConnection(self.context, sock)
-        self._environ = self.get_environ()
-        return conn
-
     def wrap(self, sock):
         """Wrap and return the given socket, plus WSGI environ entries."""
         # pyOpenSSL doesn't perform the handshake until the first read/write
         # forcing the handshake to complete tends to result in the connection
         # closing so we can't reliably access protocol/client cert for the env
-        return sock, self._environ.copy()
+        if self.context is None:
+            self.context = self.get_context()
+        conn = SSLConnection(self.context, sock)
+
+        conn.set_accept_state()  # Tell OpenSSL this is a server connection
+        self._environ = self.get_environ()
+        return conn, self._environ
```

Comment:
Why does this stop copying the environ? It's mutable after all, we don't want it to be changed externally.